File: backend/src/config/firebase.js
/**
 * Firebase Configuration
 * File konfigurasi untuk Firebase Admin SDK
 * 
 * PERHATIAN: 
 * - JANGAN mengisi credential di file ini
 * - Gunakan environment variable untuk menyimpan credential
 * - Developer backend yang akan mengisi konfigurasi
 */

const admin = require('firebase-admin');

// Firebase configuration from environment variables
const firebaseConfig = {
  projectId: process.env.FIREBASE_PROJECT_ID,
  privateKeyId: process.env.FIREBASE_PRIVATE_KEY_ID,
  privateKey: process.env.FIREBASE_PRIVATE_KEY?.replace(/\\n/g, '\n'),
  clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
  clientId: process.env.FIREBASE_CLIENT_ID,
  authUri: process.env.FIREBASE_AUTH_URI || 'https://accounts.google.com/o/oauth2/auth',
  tokenUri: process.env.FIREBASE_TOKEN_URI || 'https://oauth2.googleapis.com/token',
};

// Initialize Firebase Admin SDK
let auth;
let db;

try {
  if (firebaseConfig.projectId && firebaseConfig.privateKey && firebaseConfig.clientEmail) {
    admin.initializeApp({
      credential: admin.credential.cert(firebaseConfig)
    });
    auth = admin.auth();
    db = admin.firestore();
  } else {
    console.warn('Firebase Admin SDK not initialized. Missing required environment variables.');
  }
} catch (error) {
  console.error('Error initializing Firebase Admin SDK:', error.message);
}

module.exports = {
  firebaseConfig,
  auth,
  db,
  admin
};

File: fastapi/src/config/firebase.py
"""
Firebase Configuration
Konfigurasi Firebase Admin SDK untuk FastAPI

PERHATIAN:
- JANGAN mengisi credential di file ini
- Gunakan environment variable untuk menyimpan credential
- Developer backend yang akan mengisi konfigurasi
"""

# TODO:
# Import firebase_admin
# import firebase_admin
# from firebase_admin import credentials, auth

# TODO:
# Inisialisasi Firebase Admin
# firebase_admin.initialize_app(
#     credentials.ApplicationDefault(),
#     options={
#         'projectId': os.getenv('FIREBASE_PROJECT_ID'),
#     }
# )

# TODO:
# Export auth instance
# auth = firebase_admin.auth

# Placeholder - akan diisi oleh developer backend
auth = None
db = None


def get_firebase_auth():
    """
    Mendapatkan Firebase Auth instance
    
    TODO:
    # Return firebase_admin.auth
    """
    return auth


def verify_id_token(id_token: str):
    """
    Verifikasi Firebase ID Token
    
    TODO:
    # Implementasi verifikasi token
    # return auth.verify_id_token(id_token)
    """
    return None


def create_custom_token(uid: str, role: str = None):
    """
    Buat custom token untuk user
    
    TODO:
    # Implementasi pembuatan custom token
    # return auth.create_custom_token(uid, {'role': role})
    """
    return None


def generate_password_reset_link(email: str):
    """
    Generate link reset password
    
    TODO:
    # Implementasi generate reset link
    # return auth.generate_password_reset_link(email)
    """
    return None